test/util/util.go

```diff
@@ -28,66 +25,76 @@ const (
 )
 
 // NewTestServer creates a new test server and returns the server and the listener listening on localhost's next available port.
-func NewTestServer(t testing.TB, log logrus.FieldLogger, cfg *config.Config, store store.Store, ca *crypto.CA, serverCerts *crypto.TLSCertificateConfig) (*server.Server, net.Listener) {
-	t.Helper()
-	require := require.New(t)
+func NewTestServer(log logrus.FieldLogger, cfg *config.Config, store store.Store, ca *crypto.CA, serverCerts *crypto.TLSCertificateConfig) (*server.Server, net.Listener, error) {
```

Comment:
I also made them independent of the testing library to allow usability from ginkgo or testify as wanted.

Reply:
That makes sense but I believe Ginkgo can get that interface via `t := GinkgoT()` I guess I don't feel overly strong here but I was trying to ensure they are only ever used in tests.